sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -392,90 +392,87 @@ class FilesElement extends Polymer.Element {
    */
   async _upload() {
```

Comment:
Why is all of this logic in the UI layer? Really seems like it should be at a lower level.

Tests?